.github/workflows/lychee_url_checker.yml

```
@@ -0,0 +1,27 @@
+name: Check URLs with Lychee
+
+on:
+  push:
```

I think we can get rid of the `push` trigger. This would run the Action twice on a PR that is created from a branch of the main PyBaMM repository (once when the commit is "pushed" to the branch on PyBaMM, and the second time when a "pull request" is created from this branch). 

Or, maybe even better, we can run it on `push`, but only when the `push` is made to either `main` or `develop` (that is when a PR is merged, as pushing directly to `main` and `develop` is not allowed) -

```suggestion
  push:
    branches:
      - main
      - develop
```

---

Good idea, though any failing links at PRs will then be caught only when the PR is being merged

---

Yes, both options sound good.